eventually/vote/views.py

```diff
@@ -0,0 +1,337 @@
+"""
+    Views module
+    ============
+"""
+from django.http import JsonResponse, HttpResponse
+from django.views.generic.base import View
+from event.models import Event
+from team.models import Team
+from vote.models import Vote, Answer
+from utils.validators import string_validator
+
+
+class VoteView(View):
+    """VoteView handles GET, POST, PUT, DELETE requests"""
+
+    def get(self, request, team_id=None, event_id=None, vote_id=None):  # pylint: disable=unused-argument
+        """
+        Method that handles GET request.
+
+        :param request: the accepted HTTP request
+        :type request: HttpRequest object
+
+        :param team_id: id of the certain team
+        :type team_id: int
+
+        :param event_id: id of the certain event
+        :type event_id: int
+
+        :param vote_id: id of the certain vote
+        :type vote_id: int
+
+        :return: HttpResponse with status 400,404 or 200
+        """
+        user = request.user
+        team = Team.get_by_id(team_id)
+
+        if not vote_id:
+            return HttpResponse(status=400)
+
+        members = [user.id for user in team.members.all()]
+        if user.id not in members:
+            return HttpResponse(status=400)
+
+        vote = Vote.get_by_id(vote_id)
+        if not vote:
+            return HttpResponse(status=404)
+
+        vote.to_dict()
+        return HttpResponse(vote, status=200)
+
+    def post(self, request, team_id=None, event_id=None):
+        """
+        Method that handles POST request.
+
+        :param request: the accepted HTTP request
+        :type request: HttpRequest object
+
+        :param team_id: id of the certain team
+        :type team_id: int
+
+        :param event_id: id of the certain event
+        :type event_id: int
+
+        :return: HttpResponse with status 400,404 or 201
+        """
+        event = Event.get_by_id(event_id)
+        if not event:
+            return HttpResponse(status=404)
+
+        data = request.body
+        if not data:
```

Comment:
has this view provide some data validation?

Reply:
done